staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -271,6 +272,12 @@ func patchResource(
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
 
+			// Strategic merge patch is not supported for unstructured objects.
+			// Don't calculate strategic merge patch for them. Instead, return early indicating a conflict.
+			if _, ok := currentObject.(*unstructured.Unstructured); ok {
```

Comment:
I don't really like this approach myself...since it treats this like a special case. We wanted to avoid such special cases when moving for TPRs to CRDs and this sort of moves in the opposite direction again.

Any ideas on how to avoid this method then?

Reply:
Maybe wrapping the call to the SMP with something that just converts "smp not supported" errors into "conflict".

Reply:
`PatchResource` now has the supported patch types... could pass that into `patchResource` and only continue here if SMP is in the list of supported types

Reply:
👍 on it.